src/services/orderService.ts

```diff
@@ -0,0 +1,159 @@
+const db = require('../Database/Models/index.ts');
+
+interface IProduct {
+    id: number;
+    name: string;
+    sub_title: string;
+    model: string;
+    description: string;
+    price: number;
+    stock_quantity: number;
+}
+  
+interface IOrder {
+    id: number;
+    order_number: number;
+    total_amount: number;
+    order_date: string;
+    status: string;
+    payment_method: string;
+}
+
+interface IDiscount {
+    id: number;
+    percentage: number;
+    start_date: string;
+    end_date: string;
+    is_valid: boolean;
+}
+
+interface IOrderItems {
+    id: number;
+    price: number;
+    quantity: number;
+    createdAt: Date;
+    updatedAt: Date;
+    product_id: number;
+    order_id: number;
+}
+
+export const createOrder = async (userID: number,transaction = null) => {
+    try {
+        let orderNumber = generateOrderNumber()
```

Comment:
what if you generated a number that was already created ? you should generate another id if an order with the same number exists